### PR TITLE
discrim_net: Fix accidental AIRL no-op softmax

### LINKED ISSUE #138: Strange log_softmax in AIRL policy train reward
Firstly, thanks for this nice implementations here!
I wonder if you can help me to understand log_softmax part in **DiscrimNetAIRL.build_policy_train_reward**.
```
R(s,a) = logD(s,a) - log(1-D(s,a)) = f(s,a) - log π(a|s)
```
When it comes to the comments above, I got it.

But the implemented code here:
```python
self._log_D = tf.nn.log_softmax(self.reward_net.reward_output_train)
self._log_D_compl = tf.nn.log_softmax(self.log_policy_act_prob_ph)
return self._log_D - self.entropy_weight * self._log_D_compl
```
Why don't you just return these:
```python
return self.reward_net.reward_output_train - self.entropy_weight * self.log_policy_act_prob_ph
```

For now, log_softmax looks to add some extra term like `-(logsumexp(f(s,a)) - logsumexp(log π(a|s)))`. Is this some reducing variance trick or else?
Please shed a light on me.

### PR DESCRIPTION
**Pull request by shwang**
Resolves #138

### CODE REVIEW

**Comment by AdamGleave** [comment]
Also would be good to have a quick before-and-after benchmark. Nothing fancy: can be in just one or two environments. Would be nice to know how much this bug actually mattered...

**Comment by qxcv** [comment]
I can't tell at a glance whether this is correct. Why not just use `self.reward_net.reward_output_train - self.entropy_weight * self.log_policy_act_prob_ph`?

**Comment by shwang** [comment]
> I'm also confused as to why there are `log_softmax` calls in here at all. Isn't this a binary classifier?

I didn't realize this also decomposed to train_reward + ent_bonus back when I coded this up. `log_softmax` seemed like a quick way to get a logsumexp without numerical instability.

**Comment by shwang** [comment]
Using the direct approach now, I'm running a before-and-after comparison.

**Comment by AdamGleave** [comment]
should we merge this?

**Comment by shwang** [comment]
> should we merge this?

Sounds good, I'll post the before-and-after plots in here later today for reference.
